homeassistant/components/velux/light.py

```diff
@@ -0,0 +1,54 @@
+"""Support for Velux lights."""
+from pyvlx import Intensity, LighteningDevice
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    SUPPORT_BRIGHTNESS,
+    LightEntity,
+)
+
+from . import DATA_VELUX, VeluxEntity
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up light(s) for Velux platform."""
+    entities = []
+    for node in hass.data[DATA_VELUX].pyvlx.nodes:
+        if isinstance(node, LighteningDevice):
+            entities.append(VeluxLight(node))
+    async_add_entities(entities)
+
+
+class VeluxLight(VeluxEntity, LightEntity):
+    """Representation of a Velux light."""
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    @property
+    def brightness(self):
+        """Return the current brightness."""
+        brightness = int((100 - self.node.intensity.intensity_percent) * 255 / 100)
+        return brightness
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return not self.node.intensity.off and self.node.intensity.known
+
+    async def async_turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            intensity_percent = int(100 - kwargs[ATTR_BRIGHTNESS] / 255 * 100)
+            await self.node.set_intensity(
+                Intensity(intensity_percent=intensity_percent),
+                wait_for_completion=False,
+            )
+        else:
+            await self.node.turn_on(wait_for_completion=False)
```

Comment:
Why would you not wait for completion?

Reply:
All other calls to the velux API in the whole integration are `wait_for_completion=False`, for example all the calls in `cover.py` and `scene.py`. Not sure why. I did try with `wait_for_complete=True` a few months ago when I was developing this feature, but did not succeed in getting this to work, so just did the same as the original author of the integration.
Will have another go at it when I am testing the integration a final time.

Reply:
btw I found the PR where things has changed back in 2019 https://github.com/home-assistant/core/pull/21558. This caused home assistant to lock up somehow. The two core developers of this integration and pyvlx API agreed this change was ok. I will have another try, but if I also experience flaky issues I think we are ok to leave it the way it is.

Reply:
Changed to `wait_for_complete=True` and this seems to work correctly now.